miner/algo_common.go

```diff
@@ -60,78 +66,59 @@ func (e *lowProfitError) Error() string {
-	// is 10 (i.e. 10%), then the minimum effective gas price included in the same bucket as the top transaction
-	// is (1000 * 10%) = 100 wei.
-	PriceCutoffPercent int
-}
-
-type chainData struct {
-	chainConfig *params.ChainConfig
-	chain       *core.BlockChain
-	blacklist   map[common.Address]struct{}
-}
-
-type environmentDiff struct {
-	baseEnvironment *environment
-	header          *types.Header
-	gasPool         *core.GasPool  // available gas used to pack transactions
-	state           *state.StateDB // apply state changes here
-	newProfit       *big.Int
-	newTxs          []*types.Transaction
-	newReceipts     []*types.Receipt
-}
-
-func newEnvironmentDiff(env *environment) *environmentDiff {
-	gasPool := new(core.GasPool).AddGas(env.gasPool.Gas())
-	return &environmentDiff{
-		baseEnvironment: env,
-		header:          types.CopyHeader(env.header),
-		gasPool:         gasPool,
-		state:           env.state.Copy(),
-		newProfit:       new(big.Int),
+type (
+	algorithmConfig struct {
+		// DropRevertibleTxOnErr is used when a revertible transaction has error on commit, and we wish to discard
+		// the transaction and continue processing the rest of a bundle or sbundle.
+		// Revertible transactions are specified as hashes that can revert in a bundle or sbundle.
+		DropRevertibleTxOnErr bool
+		// EnforceProfit is true if we want to enforce a minimum profit threshold
+		// for committing a transaction based on ProfitThresholdPercent
+		EnforceProfit bool
+		// ExpectedProfit should be set on a per-transaction basis when profit is enforced
+		ExpectedProfit *big.Int
+		// ProfitThresholdPercent is the minimum profit threshold for committing a transaction
+		ProfitThresholdPercent int // 0-100, e.g. 70 means 70%
+		// PriceCutoffPercent is the minimum effective gas price threshold used for bucketing transactions by price.
+		// For example if the top transaction in a list has an effective gas price of 1000 wei and PriceCutoffPercent
+		// is 10 (i.e. 10%), then the minimum effective gas price included in the same bucket as the top transaction
+		// is (1000 * 10%) = 100 wei.
+		PriceCutoffPercent int
+		// EnableMultiTxSnap is true if we want to use multi-transaction snapshot for committing transactions,
+		// which reduce state copies when reverting failed bundles (note: experimental)
+		EnableMultiTxSnap bool
+	}
+
+	chainData struct {
+		chainConfig *params.ChainConfig
+		chain       *core.BlockChain
+		blacklist   map[common.Address]struct{}
+	}
+
+	// BuildBlockFunc is the function signature for building a block
+	BuildBlockFunc func(
+		simBundles []types.SimulatedBundle,
+		simSBundles []*types.SimSBundle,
+		transactions map[common.Address]types.Transactions) (*environment, []types.SimulatedBundle, []types.UsedSBundle)
+
+	// CommitTxFunc is the function signature for committing a transaction
+	CommitTxFunc func(*types.Transaction, chainData) (*types.Receipt, int, error)
```

Comment:
disagree but i have reverted because you're right that i should separate refactorings from the scope of this PR

please un-resolve if there are still concerns